framework/docs/src/4_get_started/9_carrot_app_tutorial.md

```diff
@@ -0,0 +1,523 @@
+that way from our `create_position` function we just have
+
+```rust
+// 3. Swap the indicated funds to match the asset0/asset1 ratio and deposit as much as possible in the pool for the given parameters
+    // 4. Create a new position
+    // 5. Store position id from create position response
+    let (swap_messages, create_position_msg) =
+        _create_position(deps.as_ref(), &env, &app, create_position_msg)?;
+```
+
+The \_create_position function can also be reused in the instantiate msg whenever the instantiator of this contract wants to also create the position within the same message.
+
+#### Swap the funds
+
+We need to Swap the indicated funds to match the asset0/asset1 ratio and deposit as much as possible in the pool for the given parameters.
+Imagine the user has 700USDC and 1300USDC and the ratio was 50%, Not doing this swap would mean that we could only deposit 700USDC/700USDT in the pool. Instead if we first swap 300USDC against 300 USDT we end up with roughly 1000USDC/1000USDT which allows us to create a position with more funds.
+The funds are not deposited by the owner to this endpoint instead the owner grants authz rights to this contract so that it can deposit funds on their behalf.
+
+```rust
+// in handlers/execute.rs
+// under fn _create_position add this
+
+pub(crate) fn _create_position(
+    deps: Deps,
+    env: &Env,
+    app: &App,
+    create_position_msg: CreatePositionMessage,
+) -> AppResult<(Vec<CosmosMsg>, SubMsg)> {
+    // Here we load
+
+
+    let CreatePositionMessage {
+        lower_tick,
+        upper_tick,
+        funds,
+        asset0,
+        asset1,
+        max_spread,
+        belief_price0,
+        belief_price1,
+    } = create_position_msg;
+
+    // 1. Swap the assets
+    let (swap_msgs, resulting_assets) = swap_to_enter_position(
+        deps,
+        env,
+        funds,
+        app,
+        asset0,
+        asset1,
+        max_spread,
+        belief_price0,
+        belief_price1,
+    )?;
+}
+```
+
+To make the swap we need to:
+
+1. Query the price
+2. calculate the amount of tokens to swap
+3. Execute the swap
+
+```rust
+#[allow(clippy::too_many_arguments)]
+pub fn swap_to_enter_position(
+    deps: Deps,
+    env: &Env,
+    funds: Vec<Coin>,
+    app: &App,
+    asset0: Coin,
+    asset1: Coin,
+    max_spread: Option<Decimal>,
+    belief_price0: Option<Decimal>,
+    belief_price1: Option<Decimal>,
+) -> AppResult<(Vec<CosmosMsg>, Vec<Coin>)> {
+    // 1. query the price
+    let price = query_price(deps, &funds, app, max_spread, belief_price0, belief_price1)?;
+    // 2. calculate the amount of tokens to swap
+    let (offer_asset, ask_asset, resulting_assets) =
+        tokens_to_swap(deps, funds, asset0, asset1, price)?;
+    // 3. Execute the swap
+    Ok((
+        swap_msg(deps, env, offer_asset, ask_asset, app)?,
+        resulting_assets,
+    ))
+}
+```
+
+##### Query price
```

Comment:
I think that we should just link to the code here (also for the tokens_to_swap function) and explain that the implementation requires some functional code that we don't have to deep-dive to understand this tutorial. I think devs would get lost in the details of this implementation otherwise.